File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        vector<pair<int,int>>p;
        for(int i=0;i<n;i++){
            p.push_back({nums[i],i});
        }
        sort(p.begin(),p.end());

        int left=0;
        int right=n-1;
        while(left<right){
            int sum=p[left].first+p[right].first;
            if(sum==target){
                return {p[left].second,p[right].second};
            }
            else if(sum<target){
                left++;
            }
            else{
                right--;
            }
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        n=len(nums)
        a={}
        for i,num in enumerate(nums):
            comp=target-num
            if comp in a:
                return [a[comp],i]
            a[num]=i    
        